core-bundle/src/Picker/UniversalPickerProvider.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Picker;
+
+use Contao\Controller;
+use Doctrine\DBAL\Connection;
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class UniversalPickerProvider implements PickerProviderInterface, DcaPickerProviderInterface, PickerMenuInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    private $menuFactory;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(FactoryInterface $menuFactory, RouterInterface $router, TranslatorInterface $translator, Connection $connection)
+    {
+        $this->menuFactory = $menuFactory;
+        $this->router = $router;
+        $this->translator = $translator;
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName(): string
+    {
+        return 'universalPicker';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUrl(PickerConfig $config): string
+    {
+        $table = $this->getTable($config->getContext());
+        $modules = $this->getModulesForTable($config->getContext());
+        $value = $config->getValue();
+
+        if (!$value) {
+            return $this->getUrlForValue($config, array_keys($modules)[0]);
```

Comment:
The case where `$modules` is empty should be handled even though that is very unlikely to happen :)

Reply:
Added exception in 319ef87f32e7fc220b67778d566e6a6e8fd91b35